betty/job/queue.py

```diff
@@ -0,0 +1,234 @@
+"""
+The job queue API.
+"""
+
+from __future__ import annotations
+
+from collections import defaultdict
+from enum import IntEnum
+from graphlib import TopologicalSorter
+from typing import (
+    TYPE_CHECKING,
+    Generic,
+    Self,
+    TypeAlias,
+    TypeVar,
+    final,
+)
+
+from betty.concurrent import AsynchronizedLock
+from betty.job import Context
+from betty.typing import internal
+
+if TYPE_CHECKING:
+    from collections.abc import (
+        AsyncIterator,
+        Awaitable,
+        Callable,
+        MutableMapping,
+        MutableSequence,
+    )
+    from types import TracebackType
+
+_ContextT = TypeVar("_ContextT", bound=Context)
+Job: TypeAlias = "Callable[[Queue[_ContextT]], Awaitable[None]]"
+
+
+@internal
+class StopWorking(Exception):
```

Comment:
Rename this so something more sensible. Introduce a single word for the concept of a queue no longer processing for whatever reason